### DISCUSSION TITLE
use rollout traits for gray deploy，workflow step dose not wait for all pod rollout ready  ，before handle next workflow step

### DISCUSSION THREAD

**Discussion by skyshaokai** [kubevela/kubevela#4009] [Q&A]
I use  topology 、override policy and deploy workflow  for multi deploy target  and  rollout for gray  deploy. 
edit image from  "oamdev/hello-world:v1"  to "oamdev/hello-world:latest"

<img width="593" alt="image" src="https://user-images.githubusercontent.com/11512967/170496114-6343a377-7f37-4b90-be8e-c08d4fe7a555.png">

**problem :**
when I use  rollout tarit ，I find that  workflow step "deploy2prod"  dose not wait for  all pod rollout ready ,  and  go to next step "deploy2default" .  Is that a question? Or is that the design. 

command line :
vela status first-vela-app -n default ; kubectl get pods -o custom-columns=CONTAINER:.spec.containers[0].name,IMAGE:.spec.containers[0].image -n prod

**get result :**

<img width="1246" alt="image" src="https://user-images.githubusercontent.com/11512967/170494647-33388326-d3a5-4131-813c-0e0c30e7742b.png">

**Comment by Reason94** [comment]
I also have this problem， anyone can help me?

**Reply by skyshaokai** [reply]
I now find that this should be a problem. When kubevela triggers the launch of the kruise-rollout , it queries whether the status of thekruise-rollout  is healthy to judge whether the cruise roll is completed. The current implementation feels that there is a problem, leading kubevela to sometimes immediately see that the cruise roll is healthy. See:
https://github.com/kubevela/catalog/blob/master/addons/kruise-rollout/definitions/kruise-rollout.cue

<img width="1235" alt="image" src="https://user-images.githubusercontent.com/11512967/187197583-4ca43c4d-8933-44e9-b895-c00d44fd4ec8.png">
